### DISCUSSION TITLE
Please add a feature to extract magnet links

### DISCUSSION THREAD

**Discussion by rutagoat** [cssnr/link-extractor#31] [Feature Requests]
Request is same as the title please can you add the magnet links in your extension.

and thank you for your extension

**Comment by smashedr** [comment]
Looks like magnet links are getting filtered out with this code:

```javascript
// Filter bad links like: javascript:void(0)
const filteredLinks = links.filter(
    (link) => link.lastIndexOf('://', 10) > 0
)
```

I just need to figure out how to best implement the feature. Either by improving the filter to include more results, or adding an option to not filter results at all. I will play with both options and see which one I like better.

Version 1.6 is already in progress. If I don't squeeze this into 1.6, I will have it in by 1.7.

**Comment by smashedr** [comment]
I think the best way is to make it an option to disable the internal filtering, This will extract all links, and if you only want magnet links, you can quick filter by `magnet`. I will try to get this in 1.6.
![20231113-222102955](https://github.com/cssnr/link-extractor/assets/6071159/b087ae7b-c8e2-46a7-9abb-27f369da224e)

**Comment by smashedr** [comment]
I pushed the update to this PR: https://github.com/cssnr/link-extractor/pull/30

Created new option `Default Filtering` which can be disabled from the Options or Popup. When disabled, lets you extract links like magnet, mailto, bitcoin, etc...

I just need to test this and I will get it out. Hopefully later today or tomorrow at latest. 

![firefox-20231114-171452119](https://github.com/cssnr/link-extractor/assets/6071159/06e3a458-dc1a-4066-ac3c-a9d5d7be8ed8)

**Comment by smashedr** [comment]
This update, version 1.6, is now Live on Firefox. I am still waiting for Google to publish 1.4 to Chrome before I can even submit 1.6. So if you are on Chrome, you will have to wait or load it as an unpacked.

Let me know if this this does not work for you. Thanks for the feedback.

**Reply by rutagoat** [reply]
Sorry for not replying to the previous posts

I just got the latest update and the add-on is working as needed i was able to scrap magnet links. Again thanks for the add-on.

**Comment by smashedr** [comment]
Glad this is working as expected. The update is now also live on Chrome so going to close this out.

Please let us know if there is anything else you would like to see added. Thanks.

**Reply by rutagoat** [reply]
thanks again for the extension it is working as needed.(Using firefox one)

I will surely request something new if i have any new ideas.
Thanks
